ratpack-core/src/main/java/ratpack/http/client/internal/ProxyChannelPoolHandler.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.http.client.internal;
+
+import io.netty.channel.Channel;
+import io.netty.channel.pool.AbstractChannelPoolHandler;
+import io.netty.handler.proxy.HttpProxyHandler;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslHandler;
+
+import javax.net.ssl.SSLException;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+
+public class ProxyChannelPoolHandler extends AbstractChannelPoolHandler {
```

Comment:
Inheritance appears to be how shared logic is handled with the ChannelPoolHanders, so this commit creates a ProxyChanelPoolHandler that the NoopXXXChannelPoolHanders extend.